homeassistant/components/hko/weather.py

```diff
@@ -0,0 +1,86 @@
+"""Support for the HKO service."""
+from homeassistant.components.weather import (
+    Forecast,
+    WeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LOCATION, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    API_CONDITION,
+    API_CURRENT,
+    API_FORECAST,
+    API_HUMIDITY,
+    API_TEMPERATURE,
+    ATTRIBUTION,
+    DOMAIN,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a HKO weather entity from a config_entry."""
+    name = config_entry.data[CONF_LOCATION]
+    unique_id = config_entry.unique_id
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([HKOEntity(name, unique_id, coordinator)], False)
+
+
+class HKOEntity(CoordinatorEntity, WeatherEntity):
+    """Define a HKO entity."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = WeatherEntityFeature.FORECAST_DAILY
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(self, name, unique_id, coordinator: DataUpdateCoordinator) -> None:
+        """Initialise the weather platform."""
+        super().__init__(coordinator)
+        self._name = name
+        self._attr_unique_id = unique_id
```

Comment:
```suggestion
        self._attr_unique_id = unique_id
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, unique_id)},
            entry_type=DeviceEntryType.SERVICE,
        )
```
Skipping the name it will take it from the config entry title